meilisearch-http/src/index/search.rs

```diff
@@ -496,18 +496,19 @@ impl<'a, A: AsRef<[u8]>> Formatter<'a, A> {
 
         tokens
             .map(|(word, token)| {
-                if format_options.highlight && token.is_word() && matcher.matches(token.text()).is_some() {
-                    let mut new_word = String::new();
-                    new_word.push_str(&self.marks.0);
-                    if let Some(match_len) = matcher.matches(token.text()) {
+                if let Some(match_len) = matcher.matches(token.text()) {
+                    if format_options.highlight && token.is_word() {
```

Comment:
```suggestion
                if format_options.highlight && token.is_word() {
                    if let Some(match_len) = matcher.matches(token.text()) {
```

Calling `matches` is more CPU intensive than checking that the token is a word or that we must highlight it.